pkg/chunk/local/fs_object_client.go

```diff
@@ -127,12 +135,33 @@ func (f *FSObjectClient) List(ctx context.Context, prefix string) ([]chunk.Stora
 }
 
 func (f *FSObjectClient) DeleteObject(ctx context.Context, objectKey string) error {
-	err := os.Remove(path.Join(f.cfg.Directory, objectKey))
-	if err != nil && os.IsNotExist(err) {
-		return chunk.ErrStorageObjectNotFound
+	filePath := filepath.Join(f.cfg.Directory, objectKey)
+
+	err := os.Remove(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return chunk.ErrStorageObjectNotFound
+		}
+		return err
 	}
 
-	return err
+	// remove the parent directory when it is empty
+	parentDir := filepath.Dir(filePath)
+	ok, err := isDirEmpty(parentDir)
+	if err != nil {
+		return err
+	}
+
+	if !ok {
+		return nil
+	}
```

Comment:
This is not necessary. `os.Remove` removes `parentDir` only if it's empty.

Also, there should be a check to avoid removing `f.cfg.Directory`  -- while not wrong, it would seem weird.